src/fastlane/release_actions/lib/fastlane/plugin/release_actions/helper/commit/parser.rb

```diff
@@ -0,0 +1,128 @@
+require 'strscan'
+require_relative 'commit'
+
+class Commit
+  # Parser is a Conventional Commits 1.0.0 compliant parser that turns commit
+  # messages into structured data. It uses StringScanner from the stdlib to
+  # search through commit messages for tokens and extracts the type, scope,
+  # subject, body, and footer from a commit message. If the parser finds a
+  # commit that isn't written as a Conventional Commmit, it will default to
+  # the type "change" and extract the subject and body.
+  #
+  # For more details on the specification, see
+  # https://www.conventionalcommits.org/en/v1.0.0/#specification.
+  class Parser
+    # FOOTER_KEY is a pattern that matches a footer element of a Conventional
+    # Commit. In the spec, commit messages can contain metadata in a footer that
+    # are defined by a key and a value. The key is in one of three legal forms:
+    #
+    # - "key: "
+    # - "key #"
+    # - "BREAKING CHANGE: " (this pattern also supports "BREAKING CHANGES: ")
+    FOOTER_KEY = /(^[\w-]+: |^[\w-]+ #|^BREAKING CHANGES?: )/
+
+    attr_reader :commit
+
+    def initialize(msg)
+      @breaking_change = false
+      @commit = Commit.new
+      @msg = StringScanner.new(msg.strip)
+    end
+
+    def parse
+      read_type_and_scope
+      read_subject
+      read_body
+      read_footer
+
+      return commit
+    end
+
+    private
+
+    attr_reader :msg
+
+    # Read the tokens that compromise the metadata prefixing the subject. [<type>[optional scope]: ]
+    def read_type_and_scope
+      commit.type = :change
+
+      # Match the tokens that compromise the metadata prefixing the subject. This
+      # must include a type such feat or fix, and can optionally include a scope
+      # and a bang (!) to indicate a breaking change. Examples:
+      #
+      # - "fix: "
+      # - "refactor(login-page): "
+      # - "feat(core)!: "
+      if msg.match?(/\w+(\([^\)]+\))?\!?: /)
+        # Read up until a bang (!), colon (:), or open parenthesis (()
+        commit.type = msg.scan(/[^!:\(]+/).to_sym
```

Comment:
We should downcase the commit type:

> The units of information that make up Conventional Commits MUST NOT be treated as case sensitive by implementors, with the exception of BREAKING CHANGE which MUST be uppercase.

Thus, `Fix`, `FIX`, and `fix` should all be treated as the same commit type.

Reply:
great catch!